### ISSUE TITLE
Cannot open a pdf generated by pdfme v4 with dynamic table data using Adobe

### ISSUE THREAD

**Issue by woodcox** [pdfme/pdfme#480]
### Describe the bug

If your default way of opening pdf's is using Adobe Acrobat Reader, any pdf generated by pdfme that contains a table with dynamic data will not open. Acrobat Reader has the following error - There was an error opening this document. The file is damaged and could not be repaired.

### To Reproduce

Go on the [pdfme playground](https://playground.pdfme.com/) and select the Invoice Template Preset. Click the Generate PDF button. This will download the pdf. Open the pdf using Adobe Acrobat Reader and the error will occur. Please note, if you select the certificate template and click the Generate PDF button. Adobe Acrobat Reader will open this pdf as normal. The error only occurs when you add a table to the template.

![image](https://github.com/pdfme/pdfme/assets/64870518/9e1accd8-745c-4abb-97cc-fab6a2356a32)

### Expected behavior

Adobe Acrobat Reader will open the pdf.

### Your Environment

```markdown
In the browser using https://playground.pdfme.com/ and opening a pdf with Adobe.
```

### Your Error Log

```shell
There is no error log as its an Adobe error message.
```

**Comment by Firephoenix25** [comment]
+1

**Comment by Jactami** [comment]
I've encountered the same issue where PDFs generated by pdfme cannot be opened with Adobe Acrobat Reader. Interestingly, the problem doesn't seem to be directly related to dynamic tables. I've pinpointed the issue to how the basePdf is defined within the template.

Using the programmatic definition of a basePDF (which is required for dynamic tables) consistently results in PDF documents that cannot be opened with Adobe Acrobat Reader:

```ts
// This won't work!
const badTemplate: Template = {
  schemas: [],     
  basePdf: {
    width: 210,
    height: 297,
    padding: [10, 10, 10, 10],
  },
}
```

On the other hand, using the default method for defining the basePdf, works as expected and the generated PDFs can be opened without any issues:

```ts
// This works fine!  
const goodTemplate: Template = {
  schemas: [],  
  basePdf: BLANK_PDF,
}
```

**Comment by ajsergeev** [comment]
I see the same issue

**Comment by ajsergeev** [comment]
If I change the way pages added to the doc in generator, output pdf works for me:

![изображение](https://github.com/pdfme/pdfme/assets/2513770/8031ce30-f5a2-4293-a640-233e9bd40a58)

**Comment by hand-dot** [comment]
This issue has been resolved in [version 4.1.0](https://github.com/pdfme/pdfme/releases/tag/4.1.0), so I'm closing it.
If you still experience problems with the latest version, please let us know.
